mirdata/datasets/haydn_op20.py

```diff
@@ -0,0 +1,375 @@
+    return score, rna_clean
+
+
+@io.coerce_to_string_io
+def load_score(fhandle: TextIO):
+    """Load haydn op20 score with annotations from a file with music21 format (music21.stream.Score).
+
+    Args:
+        fhandle (str or file-like): path to score
+
+    Returns:
+        music21.stream.Score: score in music21 format
+
+    """
+    score, rna = _split_score_annotations(fhandle)
+    return score
+
+
+def _load_key_base(fhandle, resolution):
+    """Load haydn op20 key data from a file in music21 format
+
+    Args:
+        fhandle (str or file-like): path to key annotations
+        resolution (int): the number of pulses, or ticks, per quarter note (PPQ)
+
+    Returns:
+        list: musical key data and relative time (offset (Music21Object.offset) * resolution) [(time in PPQ, local key)]
+
+    """
+    _, rna = _split_score_annotations(fhandle)
+    annotations = []
+    for offset, rn in rna:
+        time = int(round(float(offset * resolution)))
+        tonicizedKey = rn.secondaryRomanNumeralKey
+        key = tonicizedKey or rn.key
+        annotations.append({"time": time, "key": key})
+    return annotations
+
+
+@io.coerce_to_string_io
+def load_key(fhandle: TextIO, resolution=28):
+    """Load haydn op20 key data from a file
+
+    Args:
+        fhandle (str or file-like): path to key annotations
+        resolution (int): the number of pulses, or ticks, per quarter note (PPQ)
+
+    Returns:
+        KeyData: loaded key data
+
+    """
+    keys = _load_key_base(fhandle, resolution)
+    start_times, end_times, key_names = [0], [], [str(keys[0]["key"]).replace("-", "b")]
+    for ii, k in enumerate(keys):
+        if str(k["key"]).replace("-", "b") != key_names[-1]:
+            end_times.append(keys[ii]["time"] - 1)
+            start_times.append(keys[ii]["time"])
+            key_names.append(str(keys[ii]["key"]).replace("-", "b"))
+    end_times.append(keys[-1]["time"])
+    return KeyData(np.array([start_times, end_times]).astype(float).T, key_names)
+
+
+@io.coerce_to_string_io
+def load_key_music21(fhandle: TextIO, resolution=28):
+    """Load haydn op20 key data from a file in music21 format
+
+    Args:
+        fhandle (str or file-like): path to key annotations
+        resolution (int): the number of pulses, or ticks, per quarter note (PPQ)
+
+    Returns:
+        list: musical key data and relative time (offset (Music21Object.offset) * resolution) [(time in PPQ, local key)]
+
+    """
+    return _load_key_base(fhandle, resolution)
+
+
+@io.coerce_to_string_io
+def load_midi_path(fhandle: TextIO):
+    """Load path to midi file of haydn op20 musical piece
+
+    Args:
+        fhandle (str or file-like): path to score file
+
+    Returns:
+        str: midi file path
+
+    """
+    midi_path = os.path.splitext(fhandle.name)[0] + ".midi"
+    if not os.path.exists(midi_path):
+        score, _ = _split_score_annotations(fhandle)
+        score.write("midi", fp=midi_path)
+    return midi_path
```

Comment:
I still have a bit of an issue with the naming/usage of this function. I understand that you want to convert the data to midi format and I think it's a good idea.

Still, I have some concerns with this function:
It's called "load midi path". Call a function "load blabla" makes me expect the function to return data from a file - in this case a midi-like object - not a filepath string. The way the function is written right now, it's more like "convert_and_save_to_midi", and the user would still have to again load the midi data themselves.

Instead, what about calling this function "load_midi", and have it return the score data in midi format?
Something like:
```
midi_score = music21.converter.parse(fhandle.name, format="midi")
``` 
and returning this object? Then no midi_paths/saving are needed at all.

Reply:
hi @rabittl! the problem is that music21 doesnt support midi objects only io file functions :S. I think to use other library to manage it (mido/pretty_midi) is a bit overengineering. What do you think?

I fix the current name to "convert_and_save_to_midi"!!